File: app/src/main/java/com/example/raintodo/view/adapter/ViewPager2Adapter.java
package com.example.raintodo.view.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.raintodo.view.fragment.MainFragment;
import com.example.raintodo.view.fragment.UserFragment;

public class ViewPager2Adapter extends FragmentStateAdapter {
    public ViewPager2Adapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch(position) {
            case 0:
                return new MainFragment();
            case 1:
                return new UserFragment();
            default:
                return new MainFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/example/raintodo/view/adapter/ViewPager2Adapter.kt
package com.example.raintodo.view.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.raintodo.view.fragment.MainFragment
import com.example.raintodo.view.fragment.UserFragment

class ViewPager2Adapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        when (position) {
            0 -> return MainFragment()
            1 -> return UserFragment()
            else -> return MainFragment()
        }
    }

    override fun getItemCount(): Int {
        return 2
    }
}